SpatialGDK/Build/Programs/Improbable.Unreal.Scripts/DeploymentLauncher/DeploymentLauncher.cs

```diff
@@ -0,0 +1,373 @@
+                    Id = deploymentId,
+                    ProjectName = projectName
+                });
+            }
+            catch (Grpc.Core.RpcException e)
+            {
+                if (e.Status.StatusCode == Grpc.Core.StatusCode.NotFound)
+                {
+                    Console.WriteLine("<error:unknown-deployment>");
+                }
+                else
+                {
+                    throw;
+                }
+            }
+
+            return 0;
+        }
+
+        /**
+         * Prints a list of deployments within the specified project that
+         * are in state Starting or Running.
+         */
+        private static int ListDeployments(string[] args)
+        {
+            var projectName = args[1];
+
+            var deploymentServiceClient = DeploymentServiceClient.Create();
+            var listDeploymentsResult = deploymentServiceClient.ListDeployments(new ListDeploymentsRequest
+            {
+                View = ViewType.Basic,
+                ProjectName = projectName,
+                PageSize = 50
+            });
+
+            foreach (var deployment in listDeploymentsResult)
+            {
+                var status = deployment.Status;
+                if (status != Deployment.Types.Status.Starting && status != Deployment.Types.Status.Running)
+                {
+                    continue;
+                }
+
+                var overviewPageUrl = $"https://console.improbable.io/projects/{projectName}/deployments/{deployment.Name}/overview/{deployment.Id}";
+                if (deployment.Tag.Contains(SIM_PLAYER_DEPLOYMENT_TAG))
+                {
+                    Console.WriteLine($"<simulated-player-deployment> {status} {deployment.Id} {deployment.Name} {overviewPageUrl} {status}");
+                }
+                else
+                {
+                    Console.WriteLine($"<deployment> {deployment.Id} {deployment.Name} {overviewPageUrl} {status}");
+                }
+            }
+
+            return 0;
+        }
+
+        private static void ShowUsage()
+        {
+            Console.WriteLine("Usage:");
+            Console.WriteLine("DeploymentLauncher create <project-name> <assembly-name> <main-deployment-name> <main-deployment-json> <main-deployment-snapshot> [<sim-deployment-name> <sim-deployment-json> <num-sim-players>]");
+            Console.WriteLine("DeploymentLauncher stop <project-name> <deployment-id>");
+            Console.WriteLine("DeploymentLauncher list <project-name>");
+        }
+
+        private static int Main(string[] args)
+        {
+            if (args.Length == 0 ||
+                args[0] == "create" && (args.Length != 9 && args.Length != 6) ||
+                args[0] == "stop" && args.Length != 3 ||
+                args[0] == "list" && args.Length != 2)
+            {
+                ShowUsage();
+                return 1;
+            }
+
+            try
+            {
+                if (args[0] == "create")
```

Comment:
How stateless are these arguments for each command? We should probably make this as stateless as possible so that the editor does not have to hold state.

e.g. instead of `stop` specifying deployment name, `stop` could look for 'scale test' tags and find the deployment names itself.

Reply:
Good point, will change the script accordingly. That will make it easier to run bot deployments from e.g. CI as well